transformer_lens/HookedTransformer.py

```diff
@@ -504,12 +504,7 @@ def forward(
                     tokens,
                     shortformer_pos_embed,
                     attention_mask,
-                ) = self.input_to_embed(
-                    input,
-                    prepend_bos=prepend_bos,
-                    padding_side=padding_side,
-                    past_kv_cache=past_kv_cache,
-                )
+                ) = self.input_to_embed(input, past_kv_cache=past_kv_cache)
```

Comment:
`prepend_bos=prepend_bos` and `padding_side=padding_side` must not be removed from this method call to enable overriding the values.

Reply:
The body of `input_to_embed` must be wrapped with
```python3
        with utils.LocallyOverridenDefaults(
            self, prepend_bos=prepend_bos, padding_side=padding_side
        ):
```
in order to enable the default value overriding.

I've stated this in the comment below:
```
class LocallyOverridenDefaults:
    """
    Context manager that allows temporary overriding of default values within a model.
    Once the context is exited, the default values are restored.

    WARNING: This context manager must be used for any function/method that directly accesses
    default values which may be overridden by the user using the function/method's arguments,
    e.g., `model.cfg.default_prepend_bos` and `model.tokenizer.padding_side` which can be
    overriden by `prepend_bos` and `padding_side` arguments, respectively, in the `to_tokens`.
    """
```